### ISSUE TITLE
"There is data being sent, please do not operate frequently !" issue

### ISSUE THREAD

**Issue by joseaguardia** [itead/Sonoff_Devices_DIY_Tools#49]
Hello.

I've a Sonoff Mini with 3.3.0 firmware installed, the latest version of Sonoff_Devices_DIY_Tools
on a Windows 10 phisical machine.

When I run the tool, my device is detected. However, when I clic on any option (ON, OFF, INFO...) I get the error on subject. You can see  iit on the image attached.

![Captura](https://user-images.githubusercontent.com/16305835/66835671-3e24aa80-ef60-11e9-9f2f-8cafad74c0a6.PNG)

I've tried:
- Install Python 3.8
- Run the tool as admin

In the log apperars:

`刷新 eWeLink_1000982d8e._ewelink._tcp.local. 的信息 长度 382 ServiceInfo(type='_ewelink._tcp.local.', name='eWeLink_1000982d8e._ewelink._tcp.local.', address=b'\xc0\xa8\x04\x0b', port=8081, weight=0, priority=0, server='eWeLink_1000982d8e.local.', properties={b'txtvers': b'1', b'id': b'1000982d8e', b'type': b'diy_plug', b'apivers': b'1', b'seq': b'1', b'data1': b'{"switch":"off","startup":"off","pulse":"off","pulseWidth":500,"rssi":-64}'})
1000982d8e >>> {'ip': '192.168.4.11', 'port': '8081', 'switch': False, 'startup': 'off', 'pulse': False, 'pulseWidth': 500, 'rssi': -64}
在 2 列 第 0 行
run_a_dev传入设备： 1000982d8e
func_task {'info': {'1000982d8e': {'ip': '192.168.4.11', 'port': '8081', 'switch': False, 'startup': 'off', 'pulse': False, 'pulseWidth': 500, 'rssi': -64}}, 'select_name_list': ['1000982d8e'], 'command_num': 0, 'command_vrg': {'b': 'null'}}
['1000982d8e']
初始化ThreadForQT后，func_task=={'info': {'1000982d8e': {'ip': '192.168.4.11', 'port': '8081', 'switch': False, 'startup': 'off', 'pulse': False, 'pulseWidth': 500, 'rssi': -64}}, 'select_name_list': ['1000982d8e'], 'command_num': 0, 'command_vrg': {'b': 'null'}}
初始化ThreadForQT后，self. command_vrg=={'b': 'null'}
#解析出设备ID 1000982d8e
performer_func {'ip': '192.168.4.11', 'port': '8081', 'switch': False, 'startup': 'off', 'pulse': False, 'pulseWidth': 500, 'rssi': -64}
发送数据： http://192.168.4.11:8081/zeroconf/switch {'deviceid': '1000982d8e', 'data': {'switch': 'on'}}
正在发送 {"deviceid": "1000982d8e", "data": {"switch": "on"}}
lan_ewlink_api_error
刷新 eWeLink_1000982d8e._ewelink._tcp.local. 的信息 长度 382 ServiceInfo(type='_ewelink._tcp.local.', name='eWeLink_1000982d8e._ewelink._tcp.local.', address=b'\xc0\xa8\x04\x0b', port=8081, weight=0, priority=0, server='eWeLink_1000982d8e.local.', properties={b'txtvers': b'1', b'id': b'1000982d8e', b'type': b'diy_plug', b'apivers': b'1', b'seq': b'1', b'data1': b'{"switch":"off","startup":"off","pulse":"off","pulseWidth":500,"rssi":-64}'})
1000982d8e >>> {'ip': '192.168.4.11', 'port': '8081', 'switch': False, 'startup': 'off', 'pulse': False, 'pulseWidth': 500, 'rssi': -64}`

Can you help me, please?

Regards.

**Comment by joseaguardia** [comment]
Has this error happened to anyone else?

Regards

**Comment by myroom** [comment]
Yes, i have the same problem. :(

**Comment by bamboo8** [comment]
Yes, same problem, connected to "sonoffDiy" DIY tool sees the sonoff mini, if I press the on off switch on the mini, it will show on the DIY tool, but can not control from DIY tool.
so it is only a one way communication!
so changing SSID don't work!!
please help.

**Comment by s1nbad** [comment]
Same issue

**Comment by prairietech** [comment]
After waiting almost 2 weeks for my mini to arrive, I have the same issue. Guess you get what you pay for.....any one know of another solution? If even 2 or 3 times the price, if it works, it's worth it.

**Comment by dougbaptista** [comment]
same thing here

**Comment by prairietech** [comment]
I found a work around by using http POST statements using Node Red to turn the relay on/off.

**Comment by frolfe51** [comment]
I am getting the same error, has anyone been able to solve this yet?

**Comment by peros550** [comment]
same issue here

**Comment by peros550** [comment]
Was able to overcome it by installing new (wifi) drivers on my dell laptop. 
Still the whole process is quite unstable. If for some reason I close the app and open it again it no longer finds the sonoff again and it requires the PC to be rebooted.

**Comment by parad0xnasha** [comment]
+1 on getting this issue - Has anyone come up with something yet? This ticket has been open over 6mths now

**Comment by rickieyyy** [comment]
so i managed to fix this.
I used a chrome extension named RESTer to send the following:
POST: http://[IP]:8081/zeroconf/info
Body:   {"data": {}} 

if this returns data, than you are connected correctly, now you need to enable OTA update:
POST: http://[IP]:8081/zeroconf/ota_unlock
Body:   {"data": {}} 

After this, I was able to upgrade.
Credits: https://www.youtube.com/watch?v=FoJT5pJDEL4
